### PR TITLE
docs: Translate "Using TypeScript" page to Hindi

### PR DESCRIPTION
**Pull request by sohail-313**
This pull request addresses issue #168.

### Changes Proposed
Translated the "Using TypeScript" page in the React documentation to Hindi.

### Recording
https://github.com/reactjs/hi.react.dev/assets/85786620/59eefa3d-c619-4518-90ea-38d5e6c67dcd

### Note to Reviewers
No specific notes for reviewers at the moment. The changes are straightforward and align with the outlined translation task. If you have any questions or suggestions, please feel free to let me know.

### CODE REVIEW

**Comment by sohail-313** [comment]
@arshadkazmi42 can you validate this if you are free, It has been three days

**Comment by arshadkazmi42** [comment]
> @arshadkazmi42 can you validate this if you are free, It has been three days

@sohail-313 Sorry for the delay. I will get to it asap. I have been a bit occupied.

File: src/content/learn/typescript.md [hunk 1]
```diff
@@ -1,57 +1,56 @@
 ---
-title: Using TypeScript
+title: टाइपस्क्रिप्ट का उपयोग
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:2]:
```suggestion
title: TypeScript का उपयोग
```

lets not translate `TypeScript` word

File: src/content/learn/typescript.md [hunk 2]
```diff
@@ -1,57 +1,56 @@
...
 re: https://github.com/reactjs/react.dev/issues/5960
 ---
 
 <Intro>
-
-TypeScript is a popular way to add type definitions to JavaScript codebases. Out of the box, TypeScript [supports JSX](/learn/writing-markup-with-jsx) and you can get full React Web support by adding [`@types/react`](https://www.npmjs.com/package/@types/react) and [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) to your project.
-
+टाइपस्क्रिप्ट एक पॉपुलर तरीका है जिससे जावास्क्रिप्ट कोडबेस में टाइप डेफिनीशन जोड़ी जा सकती है। टाइपस्क्रिप्ट [JSX](/learn/writing-markup-with-jsx) को समर्थन करता है, और आप अपनें परियोजना में [`@types/react`](https://www.npmjs.com/package/@types/react) and [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) जोड़कर पूर्ण रिएक्ट वेब समर्थन प्राप्त कर सकते हैं।
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:7]:
```suggestion
टाइपस्क्रिप्ट एक पॉपुलर तरीका है जिससे जावास्क्रिप्ट कोडबेस में टाइप डेफिनीशन ऐड की जा सकती है। टाइपस्क्रिप्ट [JSX](/learn/writing-markup-with-jsx) को सपोर्ट करता है, और आप अपनें प्रोजेक्ट में [`@types/react`](https://www.npmjs.com/package/@types/react) and [`@types/react-dom`](https://www.npmjs.com/package/@types/react-dom) जोड़कर पूर्ण React वेब सपोर्ट प्राप्त कर सकते हैं।
```

Comment (arshadkazmi42) [inline src/content/learn/typescript.md:7]: Add a line break at line 7

File: src/content/learn/typescript.md [hunk 3]
```diff
@@ -1,57 +1,56 @@
...
 </Intro>
 
 <YouWillLearn>
 
-* [TypeScript with React Components](/learn/typescript#typescript-with-react-components)
-* [Examples of typing with Hooks](/learn/typescript#example-hooks)
-* [Common types from `@types/react`](/learn/typescript/#useful-types)
-* [Further learning locations](/learn/typescript/#further-learning)
+* [टाइपस्क्रिप्ट के साथ रिएक्ट कंपोनेंट्स:](/learn/typescript#typescript-with-react-components)
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:12]:
```suggestion
* [टाइपस्क्रिप्ट के साथ React कौम्पोनॅन्टस](/learn/typescript#typescript-with-react-components)
```

Update whole file to use `React` and for components use `कौम्पोनॅन्टस`

File: src/content/learn/typescript.md [hunk 4]
```diff
@@ -1,57 +1,56 @@
...
+* [हुक्स के साथ टाइपिंग के उदाहरण](/learn/typescript#example-hooks)
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:13]:
```suggestion
* [Hooks के साथ टाइपिंग के उदाहरण](/learn/typescript#example-hooks)
```

lets not translate React related keywords

File: src/content/learn/typescript.md [hunk 5]
```diff
@@ -1,57 +1,56 @@
...
+* [सामान्य प्रकारें से `@types/react`](/learn/typescript/#useful-types)
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:14]:
```suggestion
* [`@types/react` से कॉमन टाइप्स](/learn/typescript/#useful-types)
```

File: src/content/learn/typescript.md [hunk 6]
```diff
@@ -1,57 +1,56 @@
...
+* [और सीखने के स्थान](/learn/typescript/#further-learning)
 
 </YouWillLearn>
 
 ## Installation {/*installation*/}
 
-All [production-grade React frameworks](/learn/start-a-new-react-project#production-grade-react-frameworks) offer support for using TypeScript. Follow the framework specific guide for installation:
+सभी [प्रोडक्शन-ग्रेड रिएक्ट फ्रेमवर्क्स](/learn/start-a-new-react-project#production-grade-react-frameworks) टाइपस्क्रिप्ट का समर्थन करते हैं। स्थापना के लिए फ्रेमवर्क विशिष्ट गाइड का पालन करें:
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:21]:
```suggestion
सभी [प्रोडक्शन-ग्रेड React फ्रेमवर्क्स](/learn/start-a-new-react-project#production-grade-react-frameworks) TypeScript को सपोर्ट करते हैं। इंस्टालेशन के लिए फ्रेमवर्क विशिष्ट गाइड का पालन करें:
```

File: src/content/learn/typescript.md [hunk 7]
```diff
@@ -1,57 +1,56 @@
...
-* [Common types from `@types/react`](/learn/typescript/#useful-types)
-* [Further learning locations](/learn/typescript/#further-learning)
+* [टाइपस्क्रिप्ट के साथ रिएक्ट कंपोनेंट्स:](/learn/typescript#typescript-with-react-components)
+* [हुक्स के साथ टाइपिंग के उदाहरण](/learn/typescript#example-hooks)
+* [सामान्य प्रकारें से `@types/react`](/learn/typescript/#useful-types)
+* [और सीखने के स्थान](/learn/typescript/#further-learning)
 
 </YouWillLearn>
 
 ## Installation {/*installation*/}
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:19]:
```suggestion
## इंस्टालेशन {/*installation*/}
```

translate titles as well

File: src/content/learn/typescript.md [hunk 8]
```diff
@@ -1,57 +1,56 @@
...
+सभी [प्रोडक्शन-ग्रेड रिएक्ट फ्रेमवर्क्स](/learn/start-a-new-react-project#production-grade-react-frameworks) टाइपस्क्रिप्ट का समर्थन करते हैं। स्थापना के लिए फ्रेमवर्क विशिष्ट गाइड का पालन करें:
+
 
 - [Next.js](https://nextjs.org/docs/pages/building-your-application/configuring/typescript)
 - [Remix](https://remix.run/docs/en/1.19.2/guides/typescript)
 - [Gatsby](https://www.gatsbyjs.com/docs/how-to/custom-configuration/typescript/)
 - [Expo](https://docs.expo.dev/guides/typescript/)
 
-### Adding TypeScript to an existing React project {/*adding-typescript-to-an-existing-react-project*/}
+###  मौजूदा रिएक्ट परियोजना में टाइपस्क्रिप्ट जोड़ना {/*adding-typescript-to-an-existing-react-project*/}
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:29]:
```suggestion
### मौजूदा React प्रोजेक्ट में TypeScript ऐड करना {/*adding-typescript-to-an-existing-react-project*/}
```

File: src/content/learn/typescript.md [hunk 9]
```diff
@@ -1,57 +1,56 @@
...
 
-To install the latest version of React's type definitions:
+लेटेस्ट वर्शन की रिएक्ट के टाइप डेफिनीशन्स को स्थापित करने के लिए:
```
Comment (arshadkazmi42) [inline src/content/learn/typescript.md:31]:
```suggestion
लेटेस्ट वर्शन React के टाइप डेफिनीशन्स को इनस्टॉल करने के लिए:
```

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
Thank you for taking time and working on this.

I have added comments till line 35. Let's get these fixed before continuing further with review.

I will also suggest making the suggested changes in whole file using find & replace will speed up the review process

**Review by arshadkazmi42 — APPROVED** [review]

**Comment by arshadkazmi42** [comment]
Merging this and we can take care of comments in a followup PR

**Pull request merged**